File: src/apps/mobile/src/api/preview3d.ts
/** 3D preview endpoints. Types mirror apps/api/app/schemas/preview3d.py. */
import { api } from './client';

export type Preview3DStatus = 'starting' | 'processing' | 'succeeded' | 'failed' | 'canceled';

export type Preview3DJob = {
  job_id: string;
  status: Preview3DStatus;
  media_id: string | null;
  glb_url: string | null;
  error: string | null;
};

export const preview3dApi = {
  start: (mediaId: string, listingId?: string) =>
    api.post<Preview3DJob>('preview3d/start', { media_id: mediaId, listing_id: listingId }),

  poll: (jobId: string) => api.get<Preview3DJob>(`preview3d/${jobId}`),
};

File: src/apps/api/app/schemas/preview3d.py
"""3D preview schemas -- phase 12."""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class Preview3DStartIn(BaseModel):
    media_id: str  # the source photo -- must be an image_raw/image_enhanced Media she owns
    # If given, the finished GLB is attached to Listing.glb_media_id
    # automatically once reconstruction succeeds -- must be a listing she owns.
    listing_id: str | None = None


class Preview3DStatusOut(BaseModel):
    job_id: str
    status: Literal["starting", "processing", "succeeded", "failed", "canceled"]
    # Populated once status == 'succeeded'.
    media_id: str | None = None
    glb_url: str | None = None
    error: str | None = None
